src/App.tsx

```diff
@@ -1,26 +1,142 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodo, getTodos, postTodo} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorType, HandleErrors } from './components/HandleErrors/HandleErrors';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { Filter } from './types/Filter';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [isError, setError] = useState<ErrorType | null>(null);
+  const [filter, setFilter] = useState<Filter>('all');
+  const [isLoad, setLoad] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('load'));
+  }, []);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos, isError]);
+
+  const handleCloseError = () => {
```

Comment:
use useCallback for handlers and timers in general -> https://semaphoreci.com/blog/react-usecallback-hook#:~:text=useCallback%20is%20most%20commonly%20used,as%20event%20handlers%20and%20timers.